patches/400_add_action_rule_status.sql

```diff
@@ -0,0 +1,18 @@
+-- ****************************************************************************
+-- RM SQL DATABASE INSERT SCRIPT
+-- ****************************************************************************
+-- Number: 400
+-- Purpose: Add a status for the Action Rules and prepopulate for existing ARs
+-- Author: Kacper Prywata, Adam Hawtin
+-- ****************************************************************************
+
+ALTER TABLE casev3.action_rule ADD action_rule_status VARCHAR(255)
+CHECK (action_rule_status in ('SCHEDULED','SELECTING_CASES','PROCESSING_CASES','COMPLETED','ERRORED'));
+
+ALTER casev3.action_rule 
+SET status = "SCHEDULED"
+WHERE hasTriggered = 'false';
+
+ALTER casev3.action_rule 
+SET status = "COMPLETED"
+WHERE hasTriggered = 'true';
```

Comment:
```suggestion
UPDATE casev3.action_rule 
SET action_rule_status = 'COMPLETED'
WHERE has_Triggered = 'true';
```

Reply:
Just had a thought do we need this if we're allowing action rule status to be nullable for now?

Reply:
We need to do it at some point, doing it in this step means we should be able to immediately follow up with an update to make it non-nullable. 

We need to be careful about using these scripts to do data migrations, but I believe this example is safe to do as it is on a table with a limited number of rows, and is a non-destructive change, only populating values in a new column.